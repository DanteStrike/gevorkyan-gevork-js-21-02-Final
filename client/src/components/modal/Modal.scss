@import "src/assets/variables";

.modal {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 666;
  display: flex;
  align-items: center;
  justify-content: center;

  background-color: var(--modal-background-color);

  &--hide {
    display: none;
  }

  &__close {
    position: absolute;
    top: 10px;
    right: 10px;
    z-index: 777;
    font-size: 16px;
    color: var(--primary-font-color);
    cursor: pointer;
    background-color: unset;
    border: none;
  }

  &__close:hover {
    opacity: 0.8;
  }

  &__close:active {
    opacity: 0.6;
  }

  &__content {
    position: relative;
    width: 95%;
    min-width: 300px;
    max-width: 800px;
    height: 95vh;
    min-height: 600px;
  }

  &__scroll-wrap {
    height: 95vh;
    max-height: 95vh;
  }
}

@media (max-width: $tabletWidth) {
  .modal {
    &__close {
      top: 5px;
      right: 5px;
      font-size: 10px;
    }
  }
}
