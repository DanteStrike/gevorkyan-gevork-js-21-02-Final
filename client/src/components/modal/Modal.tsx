import React from 'react';
import './Modal.scss';
import {Alert} from 'antd';
import {OverlayScrollbarsComponent} from 'overlayscrollbars-react';

interface IModalProps {
  isOpen?: boolean;
  children: React.ReactNode;
  onClose?: () => void;
  isError?: boolean;
  errMsg?: string | null;
}

function Modal({children, isOpen, onClose = () => {}, isError, errMsg}: IModalProps) {
  const handleClose = () => {
    onClose();
  };

  return (
    <div className={`modal ${isOpen ? `` : `modal--hide`}`.trim()}>
      <button type="button" className="modal__close" onClick={handleClose}>
        X
      </button>
      <div className="modal__content">
        <OverlayScrollbarsComponent className="modal__scroll-wrap">{children}</OverlayScrollbarsComponent>
      </div>
      {isError && <Alert message={errMsg} type="error" className="content-layout__error" closable />}
    </div>
  );
}

Modal.defaultProps = {
  isOpen: false,
  onClose: () => {},
  errMsg: ``,
  isError: false,
};

export default Modal;
