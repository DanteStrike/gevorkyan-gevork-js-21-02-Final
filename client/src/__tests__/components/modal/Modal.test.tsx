import {shallow} from 'enzyme';
import React from 'react';
import Modal from '../../../components/modal/Modal';

describe(`Component Modal should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(
      <Modal>
        <div id="content" />
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot error test`, () => {
    const wrapper = shallow(
      <Modal isError errMsg="errMSG">
        <div id="content" />
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe(`Component Modal should work correctly`, () => {
  it(`Open/Close test`, () => {
    const closeHandle = jest.fn();

    const wrapper = shallow(
      <Modal onClose={closeHandle}>
        <div id="content" />
      </Modal>
    );
    expect(wrapper.find(`.modal.modal--hide`)).toHaveLength(1);
    wrapper.setProps({isOpen: true});
    expect(wrapper.find(`.modal.modal--hide`)).toHaveLength(0);
    wrapper.find(`button`).simulate(`click`);
    expect(closeHandle).toHaveBeenCalledTimes(1);
  });
});
