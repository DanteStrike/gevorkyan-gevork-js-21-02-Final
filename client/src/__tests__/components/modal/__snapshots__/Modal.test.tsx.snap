// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Modal should render correctly snapshot default test 1`] = `
<div
  className="modal modal--hide"
>
  <button
    className="modal__close"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
  <div
    className="modal__content"
  >
    <OverlayScrollbarsComponent
      className="modal__scroll-wrap"
    >
      <div
        id="content"
      />
    </OverlayScrollbarsComponent>
  </div>
</div>
`;

exports[`Component Modal should render correctly snapshot error test 1`] = `
<div
  className="modal modal--hide"
>
  <button
    className="modal__close"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
  <div
    className="modal__content"
  >
    <OverlayScrollbarsComponent
      className="modal__scroll-wrap"
    >
      <div
        id="content"
      />
    </OverlayScrollbarsComponent>
  </div>
  <Alert
    className="content-layout__error"
    closable={true}
    message="errMSG"
    type="error"
  />
</div>
`;
